src/Microsoft.Diagnostics.Runtime/src/Utilities/Platform/LinuxFunctions.cs

```diff
@@ -9,6 +9,61 @@ namespace Microsoft.Diagnostics.Runtime
 {
     internal sealed class LinuxFunctions : PlatformFunctions
     {
+        private const string LibDlGlibc = "libdl.so.2";
+        private const string LibDl = "libdl.so";
+
+        private readonly Func<string, IntPtr> _loadLibrary;
+        private readonly Func<IntPtr, bool> _freeLibrary;
+        private readonly Func<IntPtr, string, IntPtr> _getExport;
+
+        public LinuxFunctions()
+        {
+            Type nativeLibraryType = Type.GetType("System.Runtime.InteropServices.NativeLibrary, System.Runtime.InteropServices", throwOnError: false);
+            if (nativeLibraryType != null)
+            {
+                // .NET Core 3.0+
+                _loadLibrary = (Func<string, IntPtr>)nativeLibraryType.GetMethod("Load", new Type[] { typeof(string) })?.CreateDelegate(typeof(Func<string, IntPtr>));
+                Action<IntPtr> freeLibrary = (Action<IntPtr>)nativeLibraryType.GetMethod("Free", new Type[] { typeof(IntPtr) })?.CreateDelegate(typeof(Action<IntPtr>));
+                _freeLibrary = ptr => { freeLibrary(ptr); return true; };
```

Comment:
`_freeLibrary` (field) should be `null` if `freeLibrary` (local) is `null` since it's checked for `null` below